application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelMonitor.java

```diff
@@ -0,0 +1,245 @@
+package org.togetherjava.tjbot.commands.free;
+
+import net.dv8tion.jda.api.entities.*;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+
+/**
+ * A class responsible for monitoring the status of channels and reporting on their busy/free status
+ * for use by {@link FreeCommand}
+ */
+public class ChannelMonitor {
+    // Map to store channel ID's, use Guild.getChannels() to guarantee order for display
+    private final Map<Long, ChannelStatus> channelsToMonitor;
+    private final Map<Long, Long> postStatusInChannel;
+
+
+
+    ChannelMonitor() {
+        postStatusInChannel = new HashMap<>(); // JDA required to fetch guildID
+        channelsToMonitor = new HashMap<>();
+    }
```

Comment:
ok, the meaning of the comment is:
This is created as an empty map instead of being created via stream.collect (which is obviously more natural and understandable) because the JDA is required to retrieve some of the values (guildID) to create the mapping and that is not currently available at this point but it will probably change when CommandSystem supports more events.

If you're happy with the current flow where most of the init logic for ChannelMonitor is in FreeCommand and FreeCommand calls addChannelToMonitor and  addChannelForStatus(TextChannel)  individually for every channel and the maps in ChannelMonitor are not immutable because they couldnt be generated at creation then i'm happy to inline the constructor. (and remove the comment)

But if we will be changing it when appropriate it should stay as is?
(i should probably improve the comment somehow)

Reply:
changing comment to read `// JDA required to populate map`